app/javascript/packages/form-steps/use-history-param.ts

```diff
@@ -40,24 +14,22 @@ export function getQueryParam(queryString: string, name: string): string | null
  *
  * @see https://developer.mozilla.org/en-US/docs/Web/API/History/pushState
  *
- * @param name Parameter name to sync.
- * @param initialValue Value to use as initial in absence of another value.
- *
  * @return Tuple of current state, state setter.
  */
-function useHistoryParam(
-  name: string,
-  initialValue?: string | null,
-): [any, (nextParamValue: any) => void] {
+function useHistoryParam({ basePath }: HistoryOptions = {}): [
+  string | undefined,
+  (nextParamValue: string) => void,
+] {
   const getCurrentQueryParam = () =>
-    getQueryParam(window.location.hash.slice(1), name) ?? undefined;
+    (typeof basePath === 'string'
+      ? window.location.pathname.split(basePath)[1]?.replace(/\/$/, '')
```

Comment:
Should probably trim any leading slashes as well, to handle case where `basePath = '/foo'` and current URL is `'/foo/1`. Currently, this would treat the current value as `/1` instead of `1`.

Reply:
Improved in 994d9b4a2.